neuro_san/internals/run_context/langchain/journaling/journaling_tools_agent_output_parser.py

```diff
@@ -41,81 +32,22 @@ class JournalingToolsAgentOutputParser(ToolsAgentOutputParser):
-    # is able to use JSON schema definitions to validate fields.
-    journal: Journal
-
-    # This guy needs to be a pydantic class and in order to have
-    # a non-pydantic Journal as a member, we need to do this.
-    model_config = ConfigDict(arbitrary_types_allowed=True)
-
-    def __init__(self, journal: Journal):
-        """
-        Constructor
-
-        :param journal: The journal to write messages to
-        """
-        super().__init__(journal=journal)
-
-    async def aparse_result(self, result: list[Generation], *, partial: bool = False) -> T:
+    # pylint: disable=redefined-builtin
+    async def ainvoke(self, input: str | BaseMessage,
+                      config: RunnableConfig | None = None,
+                      **kwargs: Any | None) -> T:
         """
         Comments and method signature from superclass method.
-        Async parse a list of candidate model Generations into a specific format.
-
-        The return value is parsed from only the first Generation in the result, which
-            is assumed to be the highest-likelihood Generation.
-
-        Args:
-            result: A list of Generations to be parsed. The Generations are assumed
-                to be different candidate outputs for a single model input.
-            partial: Whether to parse the output as a partial result. This is useful
-                for parsers that can parse partial results. Default is False.
-
-        Returns:
-            Structured output.
-        """
-        # Do the superclass thing
-        result = await super().aparse_result(result, partial=partial)
-
-        # By empirical observation, when the result is a list, we are getting
-        # the "Invoking" message, which is useful when wanting to know agent thoughts.
-        # There is one AgentAction instance in the list per tool invocation,
-        # and that guy's log member has the message we want to stream as
-        # agent thought..
-        if isinstance(result, List):
-            for action in result:
-                if action.log is not None and len(action.log) > 0:
-                    agent_name, params_str = self._extract_agent_and_params(action.log)
-                    # Attempt to parse params_str as a Python dict literal.
-                    # It is expected to be a string representation of a dictionary (e.g., "{'key': 'value'}").
-                    # If parsing fails, fall back to using the original string.
-                    params: Union[Dict[str, Any], str] = JsonStructureParser().parse_structure(params_str)
-                    if not params:
-                        # Fallback: treat params_str as a plain string if it's not a valid Python literal.
-                        params = params_str
-                    action_dict: Dict[str, Any] = {
-                        "invoking_start": True,
-                        "invoked_agent_name": agent_name,
-                        "params": params
-                    }
```

Comment:
@Noravee : Not entirely sure what we might be losing here with this JsonStructureParser and action_dict going away.  Thoughts?